### PR TITLE
engine: watch all k8s events (wip, feature flagged)

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

**Comment by maiamcc** [comment]
lol @jazzdan I can't add you as a reviewer for this, but uh, should be gated!

@nicks the review we're looking for right now is just whether this new code is appropriately feature flag gated so that we can merge it into master and play with it there without breaking anything.

File: internal/engine/uid_map_manager_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/testutils/output"
+)
+
+func _TestUIDMapManager(t *testing.T) {
```
Comment (jazzdan) [inline internal/engine/uid_map_manager_test.go]: Does this test not pass?

Comment (maiamcc) [reply internal/engine/uid_map_manager_test.go]: lol nope it does not

File: cmd/events/main.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package main
```
[the remark this answers is not in the document]
Comment (maiamcc) [reply cmd/events/main.go]: 🤷‍♀️ idk if it'll be useful as we try to debug? i don't have strong feelings.

File: internal/engine/uid_map_manager_test.go (hunk 1 above)
Comment (jazzdan) [reply internal/engine/uid_map_manager_test.go]: Should we get it to pass before we merge it in? Have we looked in to why it doesn't pass?

File: cmd/events/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jazzdan) [reply cmd/events/main.go]: Sweet let's leave it in then

File: internal/engine/uid_map_manager_test.go (hunk 1 above)
Comment (maiamcc) [reply internal/engine/uid_map_manager_test.go]: Ha okay, DM doesn't need this on master anytime soon so I will make this test be green

**Comment by jazzdan** [comment]
This LGTM!

**Pull request merged**